geth/txqueue/txqueue_manager_test.go

```diff
@@ -28,34 +34,53 @@ type TxQueueTestSuite struct {
 	nodeManagerMock        *common.MockNodeManager
 	accountManagerMockCtrl *gomock.Controller
 	accountManagerMock     *common.MockAccountManager
+	server                 *gethrpc.Server
+	client                 *gethrpc.Client
+	txServiceMockCtrl      *gomock.Controller
+	txServiceMock          *fake.MockFakePublicTxApi
 }
 
 func (s *TxQueueTestSuite) SetupTest() {
 	s.nodeManagerMockCtrl = gomock.NewController(s.T())
 	s.accountManagerMockCtrl = gomock.NewController(s.T())
+	s.txServiceMockCtrl = gomock.NewController(s.T())
 
 	s.nodeManagerMock = common.NewMockNodeManager(s.nodeManagerMockCtrl)
 	s.accountManagerMock = common.NewMockAccountManager(s.accountManagerMockCtrl)
+
+	s.server, s.txServiceMock = fake.NewTestServer(s.txServiceMockCtrl)
+	s.client = gethrpc.DialInProc(s.server)
+	rpclient, _ := rpc.NewClient(s.client, params.UpstreamRPCConfig{})
+	s.nodeManagerMock.EXPECT().RPCClient().Return(rpclient)
 }
 
 func (s *TxQueueTestSuite) TearDownTest() {
 	s.nodeManagerMockCtrl.Finish()
 	s.accountManagerMockCtrl.Finish()
+	s.txServiceMockCtrl.Finish()
+	s.server.Stop()
+	s.client.Close()
 }
 
 func (s *TxQueueTestSuite) TestCompleteTransaction() {
-	s.accountManagerMock.EXPECT().SelectedAccount().Return(&common.SelectedExtKey{
-		Address: common.FromAddress(TestConfig.Account1.Address),
-	}, nil)
-
-	s.nodeManagerMock.EXPECT().NodeConfig().Return(
-		params.NewNodeConfig("/tmp", params.RopstenNetworkID, true),
-	)
-
-	// TODO(adam): StatusBackend as an interface would allow a better solution.
-	// As we want to avoid network connection, we mock LES with a known error
-	// and treat as success.
-	s.nodeManagerMock.EXPECT().LightEthereumService().Return(nil, errTxAssumedSent)
+	nodeConfig, nodeErr := params.NewNodeConfig("/tmp", params.RopstenNetworkID, true)
```

Comment:
As this block repeats three times, maybe we can extract it to a separate `TxQueueTestSuite` method?